pkg/controller/cluster/main_controller.go

```diff
@@ -324,14 +324,22 @@ func (c *Controller) syncHandler(key string) error {
 
 	mi.EnsureDefaults()
 
-	svc, err := c.serviceLister.Services(mi.Namespace).Get(mi.GetHeadlessServiceName())
-	// If the headless service doesn't exist, we'll create it
+	svc, err := c.serviceLister.Services(mi.Namespace).Get(mi.GetServiceName())
+	// If the service doesn't exist, we'll create it
 	if apierrors.IsNotFound(err) {
-		glog.V(2).Infof("Creating a new Headless Service for cluster %q", nsName)
-		svc = services.NewForCluster(mi)
+		klog.V(2).Infof("Creating a new Service for cluster %q", nsName)
+		svc = services.NewServiceForCluster(mi)
 		_, err = c.kubeClientSet.CoreV1().Services(svc.Namespace).Create(ctx, svc, cOpts)
```

Comment:
If the service cannot be created should we abort the operation or its safe to continue?